lib/puppet_x/puppetlabs/cisco_ios/transport_shim.rb

```diff
@@ -301,18 +301,19 @@ class Puppet::ResourceApi::Transport::Wrapper
   attr_reader :transport, :schema
 
   def initialize(name, url_or_config_or_transport)
-    if url_or_config_or_transport.is_a? String
-      url = URI.parse(url_or_config_or_transport)
-      raise "Unexpected url '#{url_or_config_or_transport}' found. Only file:/// URLs for configuration supported at the moment." unless url.scheme == 'file'
-      raise "Trying to load config from '#{url.path}, but file does not exist." if url && !File.exist?(url.path)
-      config = self.class.deep_symbolize(Hocon.load(url.path, syntax: Hocon::ConfigSyntax::HOCON) || {})
-    elsif url_or_config_or_transport.is_a? Hash
-      config = url_or_config_or_transport
-    elsif transport_class?(name, url_or_config_or_transport)
-      @transport = url_or_config_or_transport
+    # Check if the Puppet::Transport module exists
```

Comment:
Upon re-looking, this code could just bee removed and the `@transport ||= Puppet::ResourceApi::Transport.connect(name, url_or_config_or_transport)` is enough, as this code will only ever get called in 1.6.x versions of the RSAPI anyway.